### ISSUE TITLE
owncloud.Client.file_info() is not returning None when file is not found.

### ISSUE THREAD

**Issue by ericlincc** [owncloud/pyocclient#250]
According to [line 416](https://github.com/owncloud/pyocclient/blob/72a6666d7b9d022b5ee573a9d2568a0547823351/owncloud/owncloud.py#L416), `owncloud.Client.file_info()` should return `None` if file is not found. Instead an `HTTPResponseError: HTTP error: 404` is thrown.

According to OwnCloud's [developer manual](https://doc.owncloud.com/server/developer_manual/webdav_api/files_versions.html#file-is-not-found), an `HTTP/1.1 404 Not Found` is the expected response if file is not found. Currently, an exception is thrown by [owncloud.Client._make_dav_request()](https://github.com/owncloud/pyocclient/blob/72a6666d7b9d022b5ee573a9d2568a0547823351/owncloud/owncloud.py#L1788) and it has not been caught in `owncloud.Client.file_info()`.

**Comment by landryb** [comment]
i've just stumbled upon this issue, as suddenly after a change in my infra all `file_info()` calls on non-existent files/dirs started throwing exceptions. Turns out that removing those lines in my virtualhost/nginx config (https://github.com/nextcloud/documentation/commit/daa543e4e0af354cc026ab8bded30bf65d1cedfc) started triggering this, and readding those two lines to my nginx config, `file_info()` calls properly return `None` on missing files/dirs. That's ... puzzling.

Imo, the exception should be catched/handled inside `file_info()`.

**Comment by landryb** [comment]
@PVince81 since you recently commited again to this module and made a release, can you have a look at this issue, at least acknowledging it ? thanks !

**Comment by landryb** [comment]
i've tested this against pyocclient 0.4, first test when doing file_info on a non existing file on a nginx instance with the two `error_page` lines present:
```
[12/03 10:10] landry@build.fluela:~/scratch/autoshare $../venv/bin/python3 datamisc/tests/test_occlient_bug_250_missing.py
OCS request: GET https://drive.dev.craig.fr/ocs/v1.php/cloud/capabilities {'headers': {'OCS-APIREQUEST': 'true'}}
DAV request: PROPFIND doesntexist
Headers:  {'Depth': '0'}
DAV status: 200
Exists: None
```
commented out the two lines in the nginx config, reloaded it:
```
[12/03 10:10] landry@build.fluela:~/scratch/autoshare $../venv/bin/python3 datamisc/tests/test_occlient_bug_250_missing.py
OCS request: GET https://drive.dev.craig.fr/ocs/v1.php/cloud/capabilities {'headers': {'OCS-APIREQUEST': 'true'}}
DAV request: PROPFIND doesntexist
Headers:  {'Depth': '0'}
DAV status: 404
Exception: HTTP error: 404
```
the test code is:
```
import owncloud
import sys

oc = owncloud.Client('https://drive.dev.craig.fr', debug=True)
oc.login('ncadmin','xxxx')

try:
    print("exists:{}".format(oc.file_info('doesntexist')))
except owncloud.HTTPResponseError as e:
    print("exception:{}".format(e))
    sys.exit()

oc.logout()
```

done the same test against pyocclient 0.6:
```
[12/03 10:16] landry@build.fluela:~/scratch/autoshare $sudo ../venv/bin/pip3 install pyocclient==0.6
```
that produces the same exact result, so the issue is still present.

**Comment by landryb** [comment]
of course, a workaround client side would be to wrap all `file_info` calls within `try`/`except HTTPResponseError`:
```
try:
    file = oc.file_info(path)
except owncloud.HTTPResponseError:
    file = None
```

**Comment by DeepDiver1975** [comment]
This repository is being archived and is no longer maintained. Closing this as part of archival. Thank you for your contribution.
